contrib/libmetrohash/src/platform.h

```diff
@@ -33,20 +34,35 @@ inline static uint64_t read_u64(const void * const ptr)
     // so we use memcpy() which is the most portable. clang & gcc usually translates `memcpy()` into a single `load` instruction
     // when hardware supports it, so using memcpy() is efficient too.
     memcpy(&result, ptr, sizeof(result));
+
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
TL;DR: CityHash does the same thing, essentially (https://github.com/google/cityhash/blob/master/src/city.cc#L112).

The intention behind this change is to make the resulting hash for a given binary input the same on both platforms. Making changes under `src` would be more of a workaround, because the hash function would end up producing the same output for different binary inputs. Also, implementing it for all possible use-cases would be akin to a nightmare.

With that being said, I can certainly make steps to contribute the changes to the original repository.

Reply:
It seems that the original [repository](https://github.com/jandrewrogers/MetroHash) has been abandoned for around 5 years and the [maintainer](https://github.com/jandrewrogers) hasn't made any public contributions since 2019. In light of this, would you be willing to accept these changes?